### PR TITLE
Add simple lookup test #160

### PR DESCRIPTION
**Pull request by TobiasNx**

### CODE REVIEW

**Comment by TobiasNx** [comment]
@blackwinter can you tell me how to refrence the external mapfile for the integration test. I only receive errors at the moment.

**Comment by blackwinter** [comment]
> `lookup("type", "src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithExternalTsvFile/mapfile.tsv", "sep_char":"\t")`
> `put_filemap("metafix/src/test/resources/org/metafacture/metafix/integration/script/fromJson/toJson/put_fileMap/mapfile.tsv", "map", "sep_char":"\t")`

You got it right the first time. The tests are executed from the `metafix/` directory, so all paths have to be relative to that directory.

But you can shorten it to a path relative to the Fix file's directory:

```perl
lookup("type", "./mapfile.tsv", "sep_char": "\t")
put_filemap("./mapfile.tsv", "map", "sep_char": "\t")
```

**Comment by blackwinter** [comment]
Please have a look at the failing build first.

**Comment by TobiasNx** [comment]
> Please have a look at the failing build first.

The map files seem to have a trailing whitespace. The integration test did not return this as an error.

**Comment by blackwinter** [comment]
Could you resolve the following `.editorconfig` violations:

```
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjects/input.json:
 2: Wrong amount of left-padding spaces(want multiple of 2)
 3: Wrong amount of left-padding spaces(want multiple of 2)
 4: Wrong amount of left-padding spaces(want multiple of 2)
 5: Wrong amount of left-padding spaces(want multiple of 2)
 6: Wrong amount of left-padding spaces(want multiple of 2)
 7: Wrong amount of left-padding spaces(want multiple of 2)
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjects/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjectsWithListBind/input.json:
 2: Wrong amount of left-padding spaces(want multiple of 2)
 3: Wrong amount of left-padding spaces(want multiple of 2)
 4: Wrong amount of left-padding spaces(want multiple of 2)
 5: Wrong amount of left-padding spaces(want multiple of 2)
 6: Wrong amount of left-padding spaces(want multiple of 2)
 7: Wrong amount of left-padding spaces(want multiple of 2)
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjectsWithListBind/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
 11: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfStrings/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMap/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapWithoutMatch/test.fix:
 8: Wrong amount of left-padding spaces(want multiple of 2)
```

**Comment by TobiasNx** [comment]
Should be done.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjects/test.fix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+put_map("map",
+  "dog":"mammal",
+  "cat":"mammal",
+  "parrot":"bird",
+  "shark":"fish",
+  "dragon":"ficational animal",
+  "unicorn":"ficational animal")
```
Comment (blackwinter) [inline metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupInArrayOfObjects/test.fix:7]: I know, these are "only" test files, but I'd really like to not have typos in those either.

**Review by blackwinter — CHANGES_REQUESTED** [review]
In our unit tests, `lookup` tests are separate from other `method` tests. This would mean a separate area `lookup` here. WDYT? (/cc @fsteeg)

Also, my impression was that you named your tests with the method under test as prefix. That would mean `put_filemap` instead of `put_fileMap`.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "name" : "Jake",
+  "type" : "mammal"
+}
+{
+  "name" : "Blacky",
+  "type" : "bird"
+}
+{
+  "name" : "Jens",
+  "type" : "human"
```
Comment (blackwinter) [inline metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json:11]: This is not expected, is it? "Not matching literal" should be deleted, right? Then this test can be enabled.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix [hunk 3]
```diff
@@ -0,0 +1,9 @@
+put_map("map",
+  "dog":"mammal",
+  "cat":"mammal",
+  "parrot":"bird",
+  "shark":"fish",
+  "dragon":"ficational animal",
+  "unicorn":"ficational animal")
+
+lookup("type", "map", delete: "true")
```
Comment (TobiasNx) [inline metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix:9]: Value should only be deleted if delete option is true.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json (hunk 2 above)
Comment (TobiasNx) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json:11]: see my comment below. this is intentional.

Comment (TobiasNx) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json:11]: They are deleted by default, they should not.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix (hunk 3 above)
Comment (blackwinter) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix:9]: Which it is in this test.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json (hunk 2 above)
Comment (blackwinter) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json:11]: > They are deleted by default, they should not.

This is a different test.

**Comment by fsteeg** [comment]
> This would mean a separate area lookup here. WDYT?

+1 for grouping them under `lookup` (I think it's both good to have more structure here, and to mirror the unit test structure).

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix (hunk 3 above)
Comment (TobiasNx) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix:9]: is default as option available or is the option just ignored? See #149

Comment (blackwinter) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/test.fix:9]: It's ignored currently, so _accidentally_ resulting in the desired behaviour.

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json (hunk 2 above)
Comment (blackwinter) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json]: > [blackwinter](https://github.com/blackwinter) assigned [TobiasNx](https://github.com/TobiasNx) and unassigned [blackwinter](https://github.com/blackwinter) [6 minutes ago](https://github.com/metafacture/metafacture-fix/pull/177#event-6162988584)[](https://github.com/TobiasNx)
> @[TobiasNx](https://github.com/TobiasNx) TobiasNx assigned [blackwinter](https://github.com/blackwinter) and unassigned [TobiasNx](https://github.com/TobiasNx) [2 minutes ago](https://github.com/metafacture/metafacture-fix/pull/177#event-6163016330)

This conversation has not been resolved, has it? That's why I assigned you.

**Comment by TobiasNx** [comment]
Deleted the file too

File: metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json (hunk 2 above)
Comment (TobiasNx) [reply metafix/src/test/resources/org/metafacture/metafix/integration/method/fromJson/toJson/lookupSimpleWithInternalMapDeleteNotMatchingLiteral/expected.json]: sorry for that

**Review by blackwinter — APPROVED** [review]

**Pull request merged**